### DISCUSSION TITLE
wrong import of HBW-LC-RGBWW-IN6-DR

### DISCUSSION THREAD

**Discussion by holzwurm83** [SukramJ/aiohomematic#1425] [Ideas]
### The problem

I need help with the HBW-LC-RGBWW-IN6-DR. I have two of this and on RaspberryMatic I don’t have any problems.

I don´t use the HBW-LC-RGBWW-IN6-DR for RGB strips. I´m useing it vor 3 white strips.

On this page is written, you can use it for 2 RGB-strips, 1 RGBWW or 3 CW+WW-strips like 6 canal mode.
https://haus-bus.secure-stores.de/?showProduct=9

I´m use it with 3 CW+WW-stripe like 6 canal mode.

My problem is now that I have only two channels (1-2) like the picture below.

![](https://community-assets.home-assistant.io/original/4X/6/3/b/63b48458bd36de0010f34a1a310802b84892c55b.png)

And four canal (4-6) looks like this picture.

![](https://community-assets.home-assistant.io/original/4X/9/0/3/9035d4f59f8030fa0edbf5a15991b3aaf36a6ce2.png)

If canal 1or 2 is on and I put canal 3 on, than 2 chance to off.
I need all canals like first picture.

So, is it possible to fix the define of HBW-LC-RGBWW-IN6-DR? If not, how I can do it?

**Comment by SukramJ** [comment]
I'm sorry, but I have to say that I have little motivation to read about these homebrew devices. The homebrew devices are very special, differ from the normal eQ3 approach, and are only used by a few users.
Unfortunately I can't do anything with the information from the link.

For me it is not clear from the existing device descriptions how one should be able to switch between the operating modes.
Either you can follow up with the information, or there is someone else who can do it.
I will therefore close the issue and convert it into a discussion/feature request.

**Comment by holzwurm83** [comment]
I anderstand your motivation regarding self-made homebrew devices. But I am using professional made homebrew from the producer of haus-bus.de. 
I asked the CEO to help us to solve the problems. He is open to find a solution. I got the xml from him. 

Which details do you need else?


[hbw_lc_rgbww_in6_dr-1.xml.zip](https://github.com/danielperna84/hahomematic/files/14228990/hbw_lc_rgbww_in6_dr-1.xml.zip)

**Comment by SukramJ** [comment]
Siehe auch [discussion](https://github.com/danielperna84/hahomematic/discussions/851)
Die statische Gerätebeschreibung lag bereits vor, was mich zur Bewertung geführt hat, das sich das "professionelle" Gerät nicht an standards hält die sonst bei eQ-3 verwendet werden.
Beim HmIP-RGBW sind z. Bsp. alle benötigten Parameter in einem Kanal gebündelt, so das die Zuordnung zu einer HA-Entität klar ist.

Beim HBW-LC-RGBWW-IN6-DR sind scheinbar mehrere Kanäle nötig.

Wie unterscheidet man RGB / RGBW / RGBWW /CW/ ... pro HA Entität?
Welche Parameter/Kanäle werden pro HA Entität für  RGB / RGBW / RGBWW /CW/ ... verwendet?
Wofür sind die Känale 15 und 16?

**Comment by holzwurm83** [comment]
Danke für den Link und sorry, dass ich nicht selbst mal gesucht haben. Kenne mich mit GitHub gar nicht aus und habe bisher nur runtergeladen. Habe mir das auch gleich mal alles durchgelesen. 

Ich habe den Eigentümer von Haus-bus.de angeschrieben und er hat mir gerade geschrieben, dass er sich hier auch direkt melden möchte.

Ich versuche das mal auf basis der CCU zu erklären. Die Kanäle sehen dort wie folgt aus und die Stripes müssen nach Funktion und Verwendung angeschlossen werden. 
![image](https://github.com/danielperna84/hahomematic/assets/126335004/0d2cd0ce-358a-44dc-8ca3-4ce82d158f9a)

Auf der CCU sind die Kanäle 7-12 alle gleich und steuern jeweils den einen Kanal und die Kanäle 13-14 jeweils die Kanäle 7-9 und 10-12 steuern. Demnach muss ein CW/WW Stripe an Kanal 7 und 8 (und folgende - insgesamt 3 x CW/WW) angeschlossen werden. Gesteuert werden diese dann auch über diese Kanäle. 

Habe ich einen RGB Stripe (können zwei angeschlossen werden) werden diese je an die Kanäle 7-9 und 10-12 angeschlossen. Gesteuert werden diese dann mit den Kanälen 13-16. 
-> (13&15)=(7-9) 
-> (14&16)=(10-12) 

Betrachten wir den aktuellen Stand In HomeAssistant werden die HA-Entität für die Kanäle 7-8 richtig erkannt. Die HA-Entität für die Kanäle 9-12 sind falsch und müssten gleich wie für Kanal 7-8 sein. 
Die HA-Entität für die Kanäle 13-14 sind nach meinem test richt und die Kanäle 15-16 definieren die Farbe und sind in der HA-Entität für 13 und 14 inkludiert. 

Demnach passen nur die Kanäle 9-12 nicht und müssten so angepasste werden wie die Kanäle 7-8. 

Wie du auch schon gefragt hast ist, das hier die richtige Beschreibung. https://haus-bus.secure-stores.de/products/rgb/LED-Stripe-Dimmer.pdf

Wenn es dir hilft kann ich das auch mal versuchen und hochladen: https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition

**Reply by SukramJ** [reply]
> Wenn es dir hilft kann ich das auch mal versuchen und hochladen:

Danke, die hab ich schon.

Lassen wir mal weg wie was physisch angeschlossen wird!

Ich fasse das mal mit meinen Worten zusammen:
1. Die Kanäle 7-12 sind gleich und arbeiten als Dimmer (PWM)
2. 7/8, 9/10, 11/12 müssen zusammen verwendet werden wenn CW/WW
3. Im RGB/RGBW/RGBWW Betrieb sind die Kanäle 7-12 egal. Dafür werden die Kanäle 13+14 zusammen mit 15+16 verwendet
4. Es gibt keinen Parameter der den jeweiligen Betriebsmodus anzeigt, so das ich PWM und CW/WW für 7-12 nicht unterscheiden kann.

Ich tendiere gerade dazu das [custom mapping](https://github.com/danielperna84/custom_homematic#device-support) zu entfernen, das damals nicht zu Ende definiert wurde. Dadurch werden dann die rohen Entitäten erstellt und angezeigt.
Wenn man dann in HA eine hübscherer Lichtsteuerung haben will, kann man dazu in HA mittels [light template](https://www.home-assistant.io/integrations/light.template/) eigene Entitäten erstellen, die als Basis die von der Integration bereitgestellten Entitäten verwenden.
Die rohen Entitäten sehen dann so aus:
![6CDC4144-9A86-47FA-9789-906B3F2FF682](https://github.com/danielperna84/hahomematic/assets/5650106/ced1ab62-1892-4068-80ef-feb34a9289af)

**Reply by holzwurm83** [reply]
Erstmal viel dank, dass du dich der Sache annimmst!!!

> 1. Die Kanäle 7-12 sind gleich und arbeiten als Dimmer (PWM)

Ja, richtig, dass sind 6 Einzeldimmer, womit man jeden Kanal einzeln als PWM Dimmer verwenden kann.

> 2. 7/8, 9/10, 11/12 müssen zusammen verwendet werden wenn CW/WW

richtig

> 3. Im RGB/RGBW/RGBWW Betrieb sind die Kanäle 7-12 egal. Dafür werden die Kanäle 13+14 zusammen mit 15+16 verwendet

richtig

> 4. Es gibt keinen Parameter der den jeweiligen Betriebsmodus anzeigt, so das ich PWM und CW/WW für 7-12 nicht unterscheiden kann.

nein gibt es nicht. Es gibt einfach nur 6 Einzeldimmer (7-12), womit man jeden Kanal einzeln als PWM Dimmer verwenden kann.
Daneben gibt's halt 2 RGB Kanäle (also 7-9 und 10-12) wo man die Farbe und Helligkeit einstellt.

Der Kanal 13 (14) z.B. steuert die Helligkeit von Kanal 7-9 (10-12). Hier ein Bild aus der CCU bei der Farbe weiß:

![image](https://github.com/danielperna84/hahomematic/assets/126335004/2f60e666-8677-420a-95c5-7652fbf570c5)

Kanal 15 (16) stellen die Farbe ein und sieht in der CCU wie folgt aus:

![image](https://github.com/danielperna84/hahomematic/assets/126335004/2c0a75d4-79d8-4029-aa13-7a5c7dabb6f0)

Änder ich die Farbe auf grün verändert ändert sich auch 7-9 (10-12)

![image](https://github.com/danielperna84/hahomematic/assets/126335004/ec70d16b-745c-40d4-9406-d17695626c30)

> Ich tendiere gerade dazu das [custom mapping](https://github.com/danielperna84/custom_homematic#device-support) zu entfernen, das damals nicht zu Ende definiert wurde. Dadurch werden dann die rohen Entitäten erstellt und angezeigt.

Ja, so ist das aktuell definitiv nicht richtig. Kanal 1-6 sind davon nicht betroffen? Die werden aktuell richtig angezeigt:
![image](https://github.com/danielperna84/hahomematic/assets/126335004/141c4bbf-bd95-45f6-9cae-019665583b75)
Steuerelemente:
![image](https://github.com/danielperna84/hahomematic/assets/126335004/bcda5074-3d08-495d-ac81-0e1dcf568127)

> Wenn man dann in HA eine hübscherer Lichtsteuerung haben will, kann man dazu in HA mittels [light template](https://www.home-assistant.io/integrations/light.template/) eigene Entitäten erstellen, die als Basis die von der Integration bereitgestellten Entitäten verwenden.

Ich habe 10Jahre fhem hinter mir und beschäftige mich jetzt seit 6 Wochen mit HomeAssistant. Auf den ersten Blick muss ich mich da erst mal rein finden, wie das mit dem [light template](https://www.home-assistant.io/integrations/light.template/) funktioniert. 
Frage dazu: Kann man das custom mapping gemäß CCU definieren und trotzdem die light template nutzen, wenn man in HA eine hübscherer Lichtsteuerung haben will? So wäre eine gewisse Funktion analog der CCU abgedeckt.

**Reply by SukramJ** [reply]
> Kann man das custom mapping gemäß CCU definieren?

Nein, ich werde das Mapping entfernen, so das es garnicht erst so aussieht, als ob man etwas Richtung HA light Entität macht. 
Alles andere wird Murks.
Die CCU Semantik hat hier nichts zu suchen. Der Sinn einer Integration ist, dass in HA alle Geräte einer Semantik folgen. Bei einer HA `light` Entität werden z.Bsp. Helligkeit, Farbe, uvm. in einer Entität umgesetzt. 
Aufgrund fehlender Unterscheidungsmöglichkeiten zwischen den Betriebsmodi macht es für diese Geräte keinen Sinn diesen Aufwand zu betreiben.
Ich denke mal Du wirst mit den rohen Entitäten und einerm Template besser fahren.

**Reply by holzwurm83** [reply]
> Ich denke mal Du wirst mit den rohen Entitäten und einerm Template besser fahren.

ok, verstehe. Dann wird das, so am sinnvollsten sein! 
Aktuell fällt es mir noch schwer da rein zu kommen. Hast du evtl. ein Bsp. wie das für CW/WW aussehen könnte? Wahrscheinlich nutze ich nicht ganz das richtige wording beim suchen.

**Reply by SukramJ** [reply]
Die Änderung kann mit 1.56.0b0 getestet werden.

Die Beta-Nutzung kann in HACS aktiviert werden (Integration/erneutes Herunterladen/Beta aktivieren)

Vor dem HA Neustart das Gerät mit Service `homematicip_local.delete_device` löschen (Entwicklerwerkzeuge -> Dienste -> ...)

**Reply by SukramJ** [reply]
Bitte mit der 1.56.0b1 testen. Gerät sollte jetzt mit mapping vorhanden sein.

ggf. vor dem HA Neustart das Gerät mit Service homematicip_local.delete_device löschen (Entwicklerwerkzeuge -> Dienste -> ...)

**Reply by holzwurm83** [reply]
Bin gerade dabei das auf dem Testsystem einzuspielen. Soll ich nach etwas speziellem schauen. Was brauchst du als Feedback?

**Reply by holzwurm83** [reply]
sieht top aus und Funktioniert soweit. 

<img width="596" alt="image" src="https://github.com/danielperna84/hahomematic/assets/126335004/20b1ecf8-c16c-41af-802a-a6fe248748f2">

<img width="335" alt="image" src="https://github.com/danielperna84/hahomematic/assets/126335004/b864f1d6-cbeb-4775-9e30-bbbce36292d6">

Bei Kanal 13 + 14 muss noch ein Fehler drin sein. Wenn ich bei 14 die Farbe ändere wind diese auch im Kanal 13 und umgekehrt auf den geglichen wert gesetzt. 

Die Helligkeit verschiebt sich auch aber nicht auf den gleichen Wert, bei 13+14 und darf davon nicht beeinflusst werden. 
<img width="590" alt="image" src="https://github.com/danielperna84/hahomematic/assets/126335004/1ef8bd7e-9cdd-4b5a-a740-340ac50eb3a1">

<img width="590" alt="image" src="https://github.com/danielperna84/hahomematic/assets/126335004/8a51e18b-7099-4246-b34d-3757947c6e14">

**Reply by SukramJ** [reply]
Danke, ich hab noch was gefunden

**Reply by holzwurm83** [reply]
soll ich das nochmal testen?

**Reply by SukramJ** [reply]
Ja, aber dazu muss ich erst noch eine neue Version erstellen.

**Reply by SukramJ** [reply]
Bitte noch mal mit der 1.56.0b2 testen.

**Reply by holzwurm83** [reply]
> Bitte noch mal mit der 1.56.0b2 testen.

hi, die version wird mir nicht mehr angezeigt.

**Reply by SukramJ** [reply]
B3?

**Reply by holzwurm83** [reply]
Ok, das passt jetzt soweit, einzig der Farbpunkt ist entweder in der Mitte, oder ganz am Rand dazwischen lässt er sich nicht positionieren. Springt immer gleich an den Rand. 

<img width="592" alt="image" src="https://github.com/danielperna84/hahomematic/assets/126335004/4013db63-336f-4eae-ab84-8bcc0ad9dd3c">

**Reply by SukramJ** [reply]
Danke für die Rückmeldung. 

> einzig der Farbpunkt ist entweder in der Mitte, oder ganz am Rand dazwischen lässt er sich nicht positionieren. Springt immer gleich an den Rand.

Das wird an der zurückgemeldeten Farbwerten liegen.

**Reply by holzwurm83** [reply]
> Das wird an der zurückgemeldeten Farbwerten liegen.

Alles klar! Vielen dank für deinen Einsatz und die Umsetzung! 

Ich habe noch einige weiter Geräte von Hausbus.de Die Funktionieren alle sehr gut. Generelle Fehler wie hier gibt es nicht. Einige Kanäle werden dich richtig dargestellt, was aber wahrscheinlich immer das sein wird. Können wir uns das auch anschauen? Soll ich dafür was neues öffen?

**Reply by SukramJ** [reply]
Einstellen kannst Du gerne. Ob, und wann ich was machen werden wir sehen.
Für jedes Gerät bitte einen eigenen Diskussionsbeitrag.

**Comment by hausbus** [comment]
Hi,
ich bin Hermann von Haus-Bus.de.
Falls es noch Fragen zu unseren Modulen oder Kanälen gibt stehe ich parat. Danke erstmal für die HA Integration.
Es gibt keinen Parameter, der den Betriebsmodus anzeigt, weil man aktuell alles gleichzeitig verwenden kann.
Also die 6 Einzelkanäle als PWM Dimmer oder eben die 2 RGB Kanäle.

**Reply by SukramJ** [reply]
Danke für die Rückmeldung.

7-12 sind ausschliesslich Dimmer?
13 (+15 Farbe) und 14(+16 Farbe) sind ausschliesslich RGB Kanäle?

Also gibt es für 7-12 nicht die Möglichkeit zusätzlich noch die Farbtemperatur zu ändern?

**Comment by hausbus** [comment]
Genau so ist es. 6 Einzelkanäle oder 2 RGB Kanäle (1 bis 3 und 4 bis 6)

________________________________

**Reply by SukramJ** [reply]
> (1 bis 3 und 4 bis 6)

Ich nehme mal an, das damit die Anschlüsse gemeint sind, und nicht die CCU Kanäle.

**Comment by hausbus** [comment]
Genau, ja
________________________________

**Reply by holzwurm83** [reply]
Hallo Herrmann,

hier ist ein weitere [Aktor](https://github.com/danielperna84/hahomematic/discussions/1438) 

Kannst du bitte mal drüber schauen und wenn ich was vergessen habe, bitte ergänzen.
